sql/proposals/proposals.go

```diff
@@ -0,0 +1,187 @@
+package proposals
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/database"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+// Get gets a proposal by a given ID.
+func Get(db sql.Executor, id types.ProposalID) (proposal *types.Proposal, err error) {
+	if rows, err := db.Exec(`
+		select 
+			ballots.signature, 
+			ballots.pubkey, 
+			ballots.ballot, 
```

Comment:
actually i am not fully correct here. hare wants to know the beacon value recorded in a proposal. note that proposal data is ephemeral and will be deleted after a while.

maybe two options
- add a field beacon
- i think its better to add a blob for the whole proposal for serving fetch. and if that's the case, might as well decode the proposal blob instead of the ballot blob.

Reply:
thinking more about it, it's fine to leave your query code as is.
but still add a BLOB for the proposal for serving fetch.